src/ol_infrastructure/applications/redash/__main__.py

```diff
@@ -0,0 +1,281 @@
+    f'redash-role-policy-{redash_environment}',
+    policy_arn=policy_stack.require_output('iam_policies')['describe_instances'],
+    role=redash_instance_role.name
+)
+
+redash_instance_profile = iam.InstanceProfile(
+    f'redash-instance-profile-{env_suffix}',
+    role=redash_instance_role.name,
+    path='/ol-data/redash-profile/'
+)
+
+redash_instance_security_group = ec2.SecurityGroup(
+    f'redash-instance-{env_suffix}',
+    description='Security group to assign to Redash application to control inter-service access',
+    tags=aws_config.merged_tags({'Name': f'redash-instance-{redash_environment}'}),
+    vpc_id=data_vpc['id'],
+)
+
+# Set up Postgres instance for Redash in RDS
+redash_db_security_group = ec2.SecurityGroup(
+    f'redash-db-access-{redash_environment}',
+    description='Access from the data VPC to the Redash database',
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            protocol='tcp',
+            from_port=5432,  # noqa: WPS432
+            to_port=5432,  # noqa: WPS432
+            security_groups=[redash_instance_security_group.id],
+            description='PostgreSQL access from Redash instances'
+        ),
+        ec2.SecurityGroupIngressArgs(
+            protocol='tcp',
+            from_port=5432,
+            to_port=5432,
+            cidr_blocks=[operations_vpc['cidr']]
+        )
+    ],
+    egress=[
+        ec2.SecurityGroupEgressArgs(
+            from_port=0,
+            to_port=0,
+            protocol="-1",
+            cidr_blocks=["0.0.0.0/0"],
+            ipv6_cidr_blocks=['::/0']
+        )
+    ],
+    tags=aws_config.merged_tags({'Name': f'redash-db-access-{redash_environment}'}),
+    vpc_id=data_vpc['id']
+)
+
+redash_db_config = OLPostgresDBConfig(
+    instance_name=f'redash-db-{redash_environment}',
+    password=redash_config.require('db_password'),
+    subnet_group_name=data_vpc['rds_subnet'],
+    security_groups=[redash_db_security_group],
+    tags=aws_config.tags,
+    db_name='redash',
+    **defaults(stack)['rds'],
+)
+redash_db = OLAmazonDB(redash_db_config)
+
+redash_db_vault_backend_config = OLVaultPostgresDatabaseConfig(
+    db_name=redash_db_config.db_name,
+    mount_point=f'{redash_db_config.engine}-redash-{redash_environment}',
+    db_admin_username=redash_db_config.username,
+    db_admin_password=redash_config.require('db_password'),
+    db_host=redash_db.db_instance.address,
+)
+redash_db_vault_backend = OLVaultDatabaseBackend(redash_db_vault_backend_config)
+
+# Set up Redis instance for Redash in Elasticache
+redis_config = Config('redis')
+redash_redis_config = OLAmazonRedisConfig(
+    encrypt_transit=True,
+    auth_token=redis_config.require('auth_token'),
+    engine_version='6.x',
+    node_group_count=3,
+    replica_count=3,
+    auto_upgrade=True,
+    cluster_description='Redis cluster for Redash tasks and caching',
+    cluster_name=f'redash-redis-{redash_environment}',
+    num_instances=3,
+    security_groups=['dummy-group'],
```

Comment:
?

Reply:
I put that there because I needed to be able to instantiate the object for referencing it in the next block where I create the security group and then override it in the config object before passing it through to the component for creation.